python/pyspark/sql/tests/pandas/test_pandas_cogrouped_map.py

```diff
@@ -165,100 +148,191 @@ def merge_pandas(lft, _):
+            " column_with_long_column_name_22, column_with_long_column_name_23,"
+            " column_with_long_column_name_24, column_with_long_column_name_25,"
+            " column_with_long_column_name_26, column_with_long_column_name_27,"
+            " column_with_long_column_name_28, column_with_long_column_name_29,"
+            " column_with_long_column_name_30, column_with_long_column_name_31,"
+            " column_with_long_column_name_32, column_with_long_column_name_33,"
+            " column_with_long_column_name_34\n",
+        )
 
+    def test_apply_in_pandas_returning_no_column_names_and_wrong_amount(self):
         def merge_pandas(lft, rgt):
             if 0 in lft["id"] and lft["id"][0] % 2 == 0:
-                return pd.DataFrame([])
+                lft[3] = 0
             if 0 in rgt["id"] and rgt["id"][0] % 3 == 0:
-                return pd.DataFrame([])
-            return pd.merge(lft, rgt, on=["id", "k"])
-
-        result = (
-            left.groupby("id")
-            .cogroup(right.groupby("id"))
-            .applyInPandas(merge_pandas, "id long, k int, v int, v2 int")
-            .sort(["id", "k"])
-            .toPandas()
+                rgt[3] = 1
+            res = pd.merge(lft, rgt, on=["id", "k"])
+            res.columns = range(res.columns.size)
+            return res
+
+        self._test_merge_error(
+            fn=merge_pandas,
+            error_class=PythonException,
+            error_message_regex="Number of columns of the returned pandas.DataFrame "
+            "doesn't match specified schema.  Expected: 4  Actual: 6  Schema: id, k, v, v2\n",
         )
 
-        left = left.toPandas()
-        right = right.toPandas()
-
-        expected = pd.merge(
-            left[left["id"] % 2 != 0], right[right["id"] % 3 != 0], on=["id", "k"]
-        ).sort_values(by=["id", "k"])
+        # with very large schema the schema is abbreviated in the error message
+        schema = "id long, k long, mean double, " + ", ".join(
+            f"column_with_long_column_name_{no} integer" for no in range(35)
+        )
 
-        assert_frame_equal(expected, result)
+        def fn(lft, _):
+            # remove column names from lft DataFrame
+            lft.columns = range(lft.columns.size)
+            return lft
 
-    def test_apply_in_pandas_returning_empty_dataframe_and_wrong_number_of_columns(self):
-        left = self.data1
-        right = self.data2
+        self._test_merge_error(
+            fn=fn,
+            output_schema=schema,
+            error_class=PythonException,
+            error_message_regex="Number of columns of the returned pandas\\.DataFrame "
+            "doesn't match specified schema\\.  Expected: 38  Actual: 3  "
+            "Schema: id, k, mean, column_with_long_column_name_0, column_with_long_column_name_1,"
+            " column_with_long_column_name_2, column_with_long_column_name_3,"
```

Comment:
I think regex match is fine. this is too long .e.g, just `column_with_long_column_name_.*` would be good enough.

Reply:
this is gone